app/views/kudo_mailer/batch_kudo_notification.html.erb

```diff
@@ -1,10 +1,10 @@
 <% content_for :message do %>
 
   <% @commentables.each_with_index do |commentable, index| %>
-  
-    <% givers_list = @kudo_givers["#{commentable.class.name}_#{commentable.id}"] %>
-    <%= givers_list.each_with_index.map {|g,i| style_bold((g == "a guest" && i == 0) ? g.capitalize : g)}.to_sentence.html_safe %> 
-    left kudos on
+    <% givers_list = @kudo_givers["#{commentable.id}"] %>
```

Comment:
> The loops I went though trying to get this to work the other way :(

The other way presumably works correctly right now so I'm nervous about changing it without reason given the issues we've had in the past. 

I think your problem might come from the fact you've tacked on the `guest_count` to the `user_kudos` key on line 32 of redis_mail_queue.rb, so you could remove that from the end of `commentable_info` in kudo_mailer after processing the guest kudos. Then the key in kudo_giver would once again be #{commentable.class.name}_#{commentable.id} and this particular line doesn't need to change.

> What happens if the work is deleted between the kudos being given and the resque job being processed. I think the right thing to do would be to exit out of the function and pretend the email had been delivered however.....

I don't think we have a problem in that area, so I would leave it alone. The "commentable" (work) is checked in kudo_mailer.rb which is called by the resque job and skips any works it can't find at that point. I assume that would avoid problems with works that were deleted in the meantime.